packages/@sanity/vision/src/components/VisionGui.js

```diff
@@ -265,7 +325,7 @@ class VisionGui extends React.PureComponent {
     })
   }
 
-  handleQueryExecution() {
+  handleQueryExecution(callback) {
```

Comment:
Where is this callback being used?

Reply:
It is being used by the listen handler to run the query before starting to listen, and was added because to avoid having to duplicate or rewrite the query code to work in tandem with the listening. After finding it odd that it doesn't show anything when you click the listen button and discussing it with @mariuslundgard we made the change, but your point about events vs results is good. I think it might simplify things if I just revert this.

Reply:
Ah, I see now, my search skills failed me. Thanks!